src/scene_battle_rpg2k3.cpp

```diff
@@ -1901,7 +1901,7 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 
 	if (source->GetType() == Game_Battler::Type_Ally) {
 		auto* sprite = static_cast<Game_Actor*>(source)->GetActorBattleSprite();
-		if (sprite && !sprite->IsIdling() && battle_action_state != BattleActionState_CBAMove && battle_action_state != BattleActionState_Animation) {
+		if (sprite && !sprite->IsIdling() && battle_action_state != BattleActionState_CBAMove && battle_action_state != BattleActionState_StartAnimation && battle_action_state != BattleActionState_CBARangedWeaponInit && battle_action_state != BattleActionState_CBARangedWeaponMove && battle_action_state != BattleActionState_Animation) {
```

Comment:
This very long ``battle_action_state`` condition could be changed into a switch-case:

```
switch (battle_action_state) {
  case BattleActionState_CBAMove:
  case BattleActionState_StartAnimation:
   etc etc etc...
    return BattleActionReturn::eWait;
  default:;
```

Reply:
Did the change now.